src/client/sound/al_extensions.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+Minetest
+Copyright (C) 2023 DS
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; ifnot, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "al_extensions.h"
+
+#include "settings.h"
+#include "util/string.h"
+#include <unordered_set>
+
+namespace sound {
+
+ALExtensions::ALExtensions(const ALCdevice *deviceHandle [[maybe_unused]])
+{
+	auto blacklist_vec = str_split(g_settings->get("sound_extensions_blacklist"), ',');
+	for (auto &s : blacklist_vec) {
+		s = trim(s);
+	}
+	std::unordered_set<std::string> blacklist;
+	blacklist.insert(blacklist_vec.begin(), blacklist_vec.end());
+
+	{
+		constexpr const char *ext_name = "AL_SOFT_direct_channels_remix";
+		bool blacklisted = blacklist.find(ext_name) != blacklist.end();
+		if (blacklisted)
+			infostream << "ALExtensions: Blacklisted: " << ext_name << std::endl;
+#ifndef AL_SOFT_direct_channels_remix
```

Comment:
Is `alIsExtensionPresent` specific to OpenAL-soft? If not, I would prefer to compile all the code paths which serves as a quick sanity check in case of code changes.

```C++
#ifdef AL_SOFT_direct_channels_remix
if (1)
#else
if (0)
#endif
{
   // yes
} else {
    // no
}
```

Reply:
>Is alIsExtensionPresent specific to OpenAL-soft?

It's not.

The reason why I've done it like this is that some extensions introduce functions that you have to look up with `alGetProcAddress`, and alext.h provides the function types. In this case you can't compile all the code paths anyway. This is what the #ifs are for, to not check the code paths that would fail.
Also, I find it cleaner if all code that uses the extension is consistently enclosed in #ifs.